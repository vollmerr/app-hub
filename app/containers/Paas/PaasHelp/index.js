import React from 'react';

import Help from '../../../components/Help';

import sections from './sections';


export class PaasHelp extends React.PureComponent {
  render() {
    return (
      <Help title={'Paas Help'} sections={sections} />
    );
  }
}


export default PaasHelp;
