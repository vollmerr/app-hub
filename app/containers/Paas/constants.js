export const GET_MANAGER_DATA_REQUEST = 'app/Paas/GET_MANAGER_DATA_REQUEST';
export const GET_MANAGER_DATA_SUCCESS = 'app/Paas/GET_MANAGER_DATA_SUCCESS';
export const GET_MANAGER_DATA_FAILURE = 'app/Paas/GET_MANAGER_DATA_FAILURE';

export const UPDATE_USERS_REQUEST = 'app/Paas/UPDATE_USERS_REQUEST';
export const UPDATE_USERS_SUCCESS = 'app/Paas/UPDATE_USERS_SUCCESS';
export const UPDATE_USERS_FAILURE = 'app/Paas/UPDATE_USERS_FAILURE';

export const GET_REPORT_DATA_REQUEST = 'app/Paas/GET_REPORT_DATA_REQUEST';
export const GET_REPORT_DATA_SUCCESS = 'app/Paas/GET_REPORT_DATA_SUCCESS';
export const GET_REPORT_DATA_FAILURE = 'app/Paas/GET_REPORT_DATA_FAILURE';

export const SET_REPORT_KEY = 'app/Paas/SET_REPORT_KEY';
export const SET_REPORT_FILTER = 'app/Paas/SET_REPORT_FILTER';

// ROLES
export const ROLES = {
  MANAGER: 'PAAS Manager',
  SECURITY: 'PAAS Security',
  HR: 'PAAS HR',
};
export const ROLES_VALUES = Object.values(ROLES);
// APPROVAL CODES
export const APPROVAL = {
  NONE: undefined,
  DENY: 0,
  APPROVE: 1,
};
// ACCESS AUTHORIZATION
export const AUTH = {
  ID: '_id',
  SID: 'sid',
  FULL_NAME: 'fullName',
  EMAIL: 'email',
  POS_NUMBER: 'posNumber',
  MANAGER_SID: 'managerSid',
  MANAGER_NAME: 'managerFullName',
  STATUS: 'status',
  LAST_MODIFIED: 'lastUpdated',
  LAST_APPROVED: 'lastApproved',
  DATE_CREATED: 'created',
  AUTH_YEAR: 'authYear',
};
// AUTH STATUS
export const STATUS = {
  ACTIVE: 'active',
  INACTIVE: 'inactive',
  NO_MANAGER: 'noManager',
};
// APP TO AUTHORIZE
export const APPS = {
  APP_1: 'app1',
  APP_2: 'app2',
  APP_3: 'app3',
  APP_4: 'app4',
};
export const APP_LIST = Object.values(APPS);
export const BUTTONS = {
  APPROVE: 'approveButton',
  DENY: 'denyButton',
};
export const REPORT = {
  APPROVED: 0,
  DENIED: 1,
  PENDING: 2,
  NO_MANAGER: 3,
};
