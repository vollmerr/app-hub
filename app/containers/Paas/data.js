import { types, mapToColumns } from 'utils/data';

import { AUTH, APPS, APP_LIST, REPORT, BUTTONS } from './constants';


export const authorization = {
  [AUTH.ID]: {
    label: 'ID',
    name: AUTH.ID,
    ariaLabel: 'ID of the authorization',
  },
  [AUTH.SID]: {
    label: 'SID',
    name: AUTH.SID,
    ariaLabel: 'SID of the Employee',
  },
  [AUTH.FULL_NAME]: {
    label: 'Full Name',
    name: AUTH.FULL_NAME,
    ariaLabel: 'Full Name of the Employee',
    minWidth: 100,
    maxWidth: 150,
    isSortedDescending: true,
  },
  [AUTH.EMAIL]: {
    label: 'Email',
    name: AUTH.EMAIL,
    ariaLabel: 'Date the acknowledgment takes effect',
  },
  [AUTH.POS_NUMBER]: {
    label: 'Position Number',
    name: AUTH.POS_NUMBER,
    ariaLabel: 'Position Number of the Employee',
  },
  [AUTH.MANAGER_SID]: {
    label: 'Manager SID',
    name: AUTH.MANAGER_SID,
    ariaLabel: 'AD SID of the Employee\'s Manager',
  },
  [AUTH.MANAGER_NAME]: {
    label: 'Manager Name',
    name: AUTH.MANAGER_NAME,
    ariaLabel: 'Full Name of the Employee\'s Manager',
  },
  [AUTH.STATUS]: {
    label: 'Status',
    name: AUTH.STATUS,
    ariaLabel: 'Status of the Authorization',
  },
  [AUTH.LAST_MODIFIED]: {
    label: 'Last Modified',
    name: AUTH.LAST_MODIFIED,
    ariaLabel: 'Date the Authorization was last modified',
    data: {
      type: types.date,
    },
  },
  [AUTH.LAST_APPROVED]: {
    label: 'Date Authorized',
    name: AUTH.LAST_APPROVED,
    ariaLabel: 'Date of last Authorization denial or approval',
    data: {
      type: types.date,
    },
  },
  [AUTH.DATE_CREATED]: {
    label: 'Date Created',
    name: AUTH.DATE_CREATED,
    ariaLabel: 'Date the Authorization was created',
    data: {
      type: types.date,
    },
  },
  [APPS.APP_1]: {
    label: 'App 1 Name',
    name: APPS.APP_1,
    ariaLabel: 'App 1 AriaLabel',
    minWidth: 80,
    maxWidth: 100,
    data: {
      render: 'authorizationToggle',
    },
  },
  [APPS.APP_2]: {
    label: 'App 2 Name',
    name: APPS.APP_2,
    ariaLabel: 'App 2 AriaLabel',
    minWidth: 80,
    maxWidth: 100,
    data: {
      render: 'authorizationToggle',
    },
  },
  [APPS.APP_3]: {
    label: 'App 3 Name',
    name: APPS.APP_3,
    ariaLabel: 'App 3 AriaLabel',
    minWidth: 80,
    maxWidth: 100,
    data: {
      render: 'authorizationToggle',
    },
  },
  [APPS.APP_4]: {
    label: 'App 4 Name',
    name: APPS.APP_4,
    ariaLabel: 'App 4 AriaLabel',
    minWidth: 80,
    maxWidth: 100,
    data: {
      render: 'authorizationToggle',
    },
  },
};


const buttons = {
  [BUTTONS.APPROVE]: {
    name: BUTTONS.APPROVE,
    ariaLabel: 'Authorize all applications for a given user',
    minWidth: 60,
    maxWidth: 60,
    notSortable: true,
    data: {
      render: 'approvalButton',
    },
  },
  [BUTTONS.DENY]: {
    name: BUTTONS.DENY,
    ariaLabel: 'Deny all applications for a given user',
    minWidth: 60,
    maxWidth: 60,
    notSortable: true,
    data: {
      render: 'denyButton',
    },
  },
};

export const currentFieldsApi = [
  AUTH.ID,
  AUTH.SID,
  ...APP_LIST,
];

const currentFields = [
  AUTH.FULL_NAME,
  ...APP_LIST,
  BUTTONS.APPROVE,
  BUTTONS.DENY,
];

const previousFields = [
  AUTH.FULL_NAME,
  ...APP_LIST,
  AUTH.LAST_APPROVED,
];

export const currentColumns = mapToColumns({ ...authorization, ...buttons }, currentFields);
export const previousColumns = mapToColumns(authorization, previousFields);


const reportFields = [
  AUTH.FULL_NAME,
  ...APP_LIST,
  AUTH.MANAGER_NAME,
  AUTH.LAST_APPROVED,
];

export const reportColumns = {
  [REPORT.APPROVED]: mapToColumns(authorization, reportFields),
  [REPORT.DENIED]: mapToColumns(authorization, reportFields),
  [REPORT.PENDING]: mapToColumns(authorization, reportFields),
  [REPORT.NO_MANAGER]: mapToColumns(authorization, reportFields),
};
