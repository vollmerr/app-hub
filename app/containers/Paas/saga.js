// import { delay } from 'redux-saga';
import { takeEvery, call, put } from 'redux-saga/effects';

import requestWithToken from 'utils/requestWithToken';
import * as C from './constants';
import * as actions from './actions';


export const base = API.PAAS;


export function* getManagerData() {
  try {
    const url = `${base}`;
    const data = yield call(requestWithToken, url);
    yield put(actions.getManagerDataSuccess(data));
  } catch (error) {
    yield put(actions.getManagerDataFailure(error));
  }
}


export function* updateUsers(action) {
  try {
    const url = `${base}`;
    const options = {
      method: 'POST',
      body: action.payload,
    };

    yield call(requestWithToken, url, options);
    yield put(actions.updateUsersSuccess(action.payload));
  } catch (error) {
    yield put(actions.updateUsersFailure(error));
  }
}


function* paasSaga() {
  yield takeEvery(C.GET_MANAGER_DATA_REQUEST, getManagerData);
  yield takeEvery(C.UPDATE_USERS_REQUEST, updateUsers);
}


export default paasSaga;
