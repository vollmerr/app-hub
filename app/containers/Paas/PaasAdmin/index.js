import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { Form } from 'react-final-form';

import { Selection } from 'office-ui-fabric-react/lib/DetailsList';

import validate from './validate';
import toJS from '../../../hocs/toJS';
import { shouldFetch } from '../../../utils/api';
import Loading from '../../../components/Loading';
import List, { handleSelectItem } from '../../../components/List';
import { FormSection, FormButtons } from '../../../components/Form';

import * as hubSelectors from '../../AppHub/selectors';

import { adminNoManagerColumns, adminAssignedManagerColumns } from '../data';
import * as selectors from '../selectors';
import * as actions from '../actions';
import * as C from '../constants';
import FieldText from '../../../components/Form/FieldText';
import FieldSelect from '../../../components/Form/FieldSelect';
import theme from '../../../utils/theme';


const style = {
  count: 2,
};

export const Wrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  width: 100%;
  padding: ${theme.hub.padding / 2}px;
  min-height: calc(
    100vh - \
    ${theme.hub.headerHeight}px
  );
`;
export const FormList = styled(List) `
  box-shadow: none;
  margin: 0;
  padding: 0;
`;


export const Section = styled.div`
  flex: ${(props) => props.flex || 1};
  display: flex;
  flex-direction: column;
  width: 100%;
  min-height: 300px;
  @media (min-width: ${theme.breakpoints.xs}px) {
    min-width: ${theme.breakpoints.xs - 40}px;
  }
`;

export class PaasAdmin extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      loading: true,
      selectedEmployee: { [C.AUTH.FULL_NAME]: 'Select Employee From list on left' }, // this will be reportData Item
      selectedInitalAssignedMananger: { [C.AUTH.FULL_NAME]: 'Select Employee From list on left' }, // this will be reportData Item
    };
    this.selectionNoManager = new Selection({
      onSelectionChanged: () => handleSelectItem(this, this.selectionNoManager, this.handleSelectNoManager),
    });
    this.selectionAssignedManager = new Selection({
      onSelectionChanged: () => handleSelectItem(this, this.selectionAssignedManager, this.handleSelectAssignedManager),
    });
  }

  async componentDidMount() {
    const { report, onGetReportDataRequest } = this.props;
    if (shouldFetch(report.lastFetched)) {
      await onGetReportDataRequest();
    }

    this.setState({ loading: false }); // eslint-disable-line
  }

  getManagerOptions = () => {
    const { reportData } = this.props;
    // Exclude the employee from the list of managers for selection
    const managerFilter = (e) => {
      const x = this.state.selectedEmployee[C.AUTH.ID];
      return e[C.AUTH.ID] !== x;
    };
    // sort the manager list alphabetically
    const managerSort = (a, b) => {
      let res = 0;
      const x = a.fullName.toLowerCase();
      const y = b.fullName.toLowerCase();
      if (x > y) {
        res = 1;
      } else {
        res = -1;
      }
      return res;
    };

    const ret = reportData.all.sort(managerSort).filter(managerFilter).map((entry) => (
      {
        key: entry[C.AUTH.ID],
        text: entry[C.AUTH.FULL_NAME],
      }));
    const blank = {
      key: '',
      text: '',
    };
    ret.unshift(blank);

    return ret;
  }

  /**
   * Handles selecting an item from a list
   */
  handleSelectNoManager = (item) => {
    const selectedInitalAssignedMananger = {
      [C.AUTH.FULL_NAME]: '',
    };
    this.setState({ selectedEmployee: item });
    this.setState({ selectedInitalAssignedMananger });
  }


  /**
   * Handles selecting an item from a list
   */
  handleSelectAssignedManager = (selectedEmployee) => {
    const selectedInitalAssignedMananger = this.props.reportData.all.find((e) => e[C.AUTH.SID] === selectedEmployee[C.AUTH.MANAGER_SID]);
    this.setState({ selectedEmployee });
    this.setState({ selectedInitalAssignedMananger });
  }


  /**
   * Handles submitting a request to the api with form values
   *
   * @param {object} values   - form values
   */
  handleSubmit = async (values) => {
    const selectedInitalAssignedMananger = this.props.reportData.all
      .find((e) => e[C.AUTH.ID] === values[C.MANAGE.MANAGER_ID]);
    const { onUpdateUserManagerRequest } = this.props;
    const user = {
      [C.MANAGE.EMPLOYEE_ID]: values[C.MANAGE.EMPLOYEE_ID],
      [C.MANAGE.MANAGER_ID]: values[C.MANAGE.MANAGER_ID],
      manager: this.props.reportData.all.find((e) => e[C.AUTH.ID] === values[C.MANAGE.MANAGER_ID]),
      employee: this.props.reportData.all.find((e) => e[C.AUTH.ID] === values[C.MANAGE.EMPLOYEE_ID]),
    };

    await onUpdateUserManagerRequest(user);
    this.setState({ selectedInitalAssignedMananger });
  }

    /**
  * Renders the form
  */
  renderForm = ({ handleSubmit, reset, submitting, pristine }) => {
    const employeeProps = {
      label: 'Employee Name:',
      name: 'employeeName',
      disabled: true,
      placeholder: 'Select employee from list',
    };

    const disableManagerDropdown =
      this.state.selectedEmployee[C.AUTH.ID] === undefined;

    const managerProps = {
      label: 'Manager:',
      name: 'manager_id',
      allowFreeform: 0,
      autoComplete: 'on',
      disabled: disableManagerDropdown,
      options: this.getManagerOptions(),
      placeholder: 'Select manager from list',
    };


    const buttonProps = {
      reset,
      disabled: pristine || submitting,
    };

    return (
      <FormSection onSubmit={handleSubmit}>
        <div>Select Employee From Left</div>
        <FieldText {...employeeProps} />
        <FieldSelect {...managerProps} />
        <FormButtons {...buttonProps} />
      </FormSection>
    );
  }

  render() {
    const { app, reportData } = this.props;
    const { loading } = this.state;

    if (app.loading || app.error || loading) {
      const loadingProps = {
        loading: app.loading || loading,
        error: app.error,
        to: '/',
      };
      return <Loading {...loadingProps} />;
    }

    const listNoManProps = {
      columns: adminNoManagerColumns,
      items: reportData[C.REPORT.NO_MANAGER],
      title: 'Employees w/o Manager',
      style,
      sortBy: [C.AUTH.FULL_NAME],
      selection: this.selectionNoManager,
    };

    const listAssignedManProps = {
      columns: adminAssignedManagerColumns,
      items: reportData[C.STATUS.ASSIGNED_MANAGER],
      style,
      title: 'Emp. w/assigned Manager',
      sortBy: [C.AUTH.FULL_NAME],
      selection: this.selectionAssignedManager,
    };
    const formProps = {
      initialValues: {
        employeeName: this.state.selectedEmployee[C.AUTH.FULL_NAME],
        employee_id: this.state.selectedEmployee[C.AUTH.ID],
        manager_id: this.state.selectedInitalAssignedMananger[C.AUTH.ID],
      },
      validate,
      render: this.renderForm,
      onSubmit: this.handleSubmit,
      subscription: { submitting: true, pristine: true },
    };


    return (
      <Wrapper>
        <Section flex={5}>
          <List {...listNoManProps} />
          <List {...listAssignedManProps} />
        </Section>
        <Section>
          <Form {...formProps} />
        </Section>
      </Wrapper>
    );
  }
}


const { object, func } = PropTypes;

PaasAdmin.propTypes = {
  app: object.isRequired,
  report: object.isRequired, // eslint-disable-line
  reportData: object,
  onGetReportDataRequest: func.isRequired, // eslint-disable-line
  onUpdateUserManagerRequest: func.isRequired,  // eslint-disable-line
};


const mapStateToProps = createStructuredSelector({
  app: hubSelectors.getApp,
  report: selectors.getReport,
  reportData: selectors.getReportData,
});

export const mapDispatchToProps = (dispatch) => ({
  onGetReportDataRequest: () => dispatch(actions.getReportDataRequest()),
  onUpdateUserManagerRequest: (user) => dispatch(actions.updateUserManagerRequest(user)),
});

const withConnect = connect(mapStateToProps, mapDispatchToProps);


export default compose(
  withConnect,
  toJS,
)(PaasAdmin);
