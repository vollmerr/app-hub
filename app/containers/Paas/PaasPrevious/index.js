import React from 'react';
import PropTypes from 'prop-types';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import toJS from '../../../hocs/toJS';
import { shouldFetch } from '../../../utils/api';
import Loading from '../../../components/Loading';
import List from '../../../components/List';
import { StyledToggle } from '../../../components/Form/FieldToggle';
import { isNull } from '../../../utils/validate';

import * as hubSelectors from '../../AppHub/selectors';

import { previousColumns } from '../data';
import * as selectors from '../selectors';
import * as actions from '../actions';
import * as C from '../constants';


export class PaasPrevious extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      columns: this.buildColumns(previousColumns),
    };
  }

  async componentDidMount() {
    const { manager, onGetManagerDataRequest } = this.props;
    // only load user data if not cached
    if (shouldFetch(manager.lastFetched)) {
      await onGetManagerDataRequest();
    }
    this.setState({ loading: false }); // eslint-disable-line
  }

  /**
   * Binds columns with a custom render function
   *
   * @param {array} columns   - column objects to bind to
   *
   * @return {array}          - columns with onRender bound
   */
  buildColumns = (columns) => (
    columns.map((col) => ({
      ...col,
      onRender: this.renderColumn(col),
    }))
  )


  /**
   * Renders a List column in a custom format
   * based off data.render prop
   *
   * @param {object} column   - column to render
   *
   * @return {JSX}            - custom rendering of column
   */
  renderColumn = (column) => {
    if (column.data && column.data.render) {
      const onRenders = {
        authorizationToggle: (item) => {
          const checked = isNull(item[column.key]) ? null : Number(item[column.key]);
          const toggleProps = {
            checked,
            onText: 'Yes',
            offText: 'No',
            isNullable: true,
            disabled: true,
            readOnly: true,
          };

          return (
            <StyledToggle {...toggleProps} />
          );
        },
      };
      return onRenders[column.data.render];
    }
    return undefined;
  }


  render() {
    const { app, managerItems } = this.props;
    const { loading, columns } = this.state;

    if (app.loading || app.error || loading) {
      const loadingProps = {
        loading: app.loading || loading,
        error: app.error,
        to: app.home.path,
      };
      return <Loading {...loadingProps} />;
    }

    const listProps = {
      columns,
      items: managerItems,
      title: 'Previous Staff Authorizations',
      sortBy: [C.AUTH.FULL_NAME],
    };

    return (
      <List {...listProps} />
    );
  }
}


const { object, array, func } = PropTypes;

PaasPrevious.propTypes = {
  app: object.isRequired,
  manager: object.isRequired,
  managerItems: array.isRequired,
  managerById: object.isRequired, // eslint-disable-line
  onGetManagerDataRequest: func.isRequired, // eslint-disable-line
};

const mapStateToProps = createStructuredSelector({
  app: hubSelectors.getApp,
  manager: selectors.getManager,
  managerItems: selectors.getManagerItems('previous'),
  managerById: selectors.getManagerById('previous'),
});

export const mapDispatchToProps = (dispatch) => ({
  onGetManagerDataRequest: () => dispatch(actions.getManagerDataRequest()),
});

const withConnect = connect(mapStateToProps, mapDispatchToProps);


export default compose(
  withConnect,
  toJS,
)(PaasPrevious);
