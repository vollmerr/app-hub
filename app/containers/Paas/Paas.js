import React from 'react';
import PropTypes from 'prop-types';
import { compose } from 'redux';

import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';
import AppContainer from 'containers/App-Container';
import { meta } from 'containers/AppHub/meta';

import reducer from './reducer';
import saga from './saga';
import routes from './routes';

const app = {
  meta: meta.paas,
  routes,
};


export class Paas extends React.PureComponent {
  render() {
    app.name = this.props.name;

    return (
      <AppContainer app={app} />
    );
  }
}


const { string } = PropTypes;

Paas.propTypes = {
  name: string.isRequired,
};

const withReducer = injectReducer({ key: 'paas', reducer });
const withSaga = injectSaga({ key: 'paas', saga });

export default compose(
  withReducer,
  withSaga,
)(Paas);
