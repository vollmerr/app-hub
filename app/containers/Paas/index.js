import React from 'react';
import { compose } from 'redux';

import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';

import { meta } from '../AppHub/meta';
import App from '../App';

import routes from './routes';
import reducer from './reducer';
import saga from './saga';


const appProps = {
  routes,
  meta: meta.paas,
  name: meta.paas.title,
};


export class Paas extends React.PureComponent {
  render() {
    const props = {
      appProps,
    };

    return (
      <App {...props} />
    );
  }
}


const withReducer = injectReducer({ key: 'paas', reducer });
const withSaga = injectSaga({ key: 'paas', saga });


export default compose(
  withReducer,
  withSaga,
)(Paas);
