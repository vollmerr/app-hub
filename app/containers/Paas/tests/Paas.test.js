import React from 'react';
import { shallow } from 'enzyme';

import AppContainer from 'containers/App-Container';

import { Paas } from '../Paas';

const props = {
  name: 'test name',
  meta: { desc: 'test desc' },
  routes: [{ name: 'route1', key: 'testHome', path: '/testpath' }, { name: 'route2', key: '2', path: '/testpath2' }],
};


describe('<Paas />', () => {
  let wrapper;
  beforeEach(() => {
    jest.resetAllMocks();
    wrapper = shallow(<Paas {...props} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have render a AppContainer', () => {
    expect(wrapper.find(AppContainer).length).toEqual(1);
  });
});
