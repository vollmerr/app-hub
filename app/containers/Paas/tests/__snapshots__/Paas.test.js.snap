// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paas /> should render correctly 1`] = `
<Connect(AppContainer)
  app={
    Object {
      "meta": Object {
        "desc": "Personal Authorization Access System",
        "keywords": "tracking, access, authorization",
        "title": "PAAS",
      },
      "name": "test name",
      "routes": Array [
        Object {
          "component": [Function],
          "exact": true,
          "key": "paasHome",
          "name": "Home",
          "path": "/paas/home",
          "roles": Array [
            "PAAS Manager",
          ],
        },
        Object {
          "component": [Function],
          "exact": true,
          "key": "paasReport",
          "name": "Report",
          "path": "/paas/report",
          "roles": Array [
            "PAAS Manager",
            "PAAS HR Admin",
            "PAAS Security Admin",
          ],
        },
      ],
    }
  }
/>
`;
